File: app/src/main/java/com/fisherman/ordertogether/MainActivity.java
package com.fisherman.ordertogether;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.navigation.NavigationView;
import com.google.android.material.snackbar.Snackbar;

public class MainActivity  extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener{

    private Toolbar toolbar;
    private DrawerLayout drawerLayout;
    private NavController navController;
    private NavigationView navigationView;
    private Snackbar snackbar;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupNavigation();
    }

    private void setupNavigation(){

        // attaching view components to view variables
        toolbar = findViewById(R.id.toolbar);
        drawerLayout = findViewById(R.id.drawer_layout);
        navigationView = findViewById(R.id.navigationView);

        // toolbar setup
        setSupportActionBar(toolbar);
//        getSupportActionBar().setIcon(R.drawable.ic_android_black_24dp);
//        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
//        getSupportActionBar().setDisplayShowHomeEnabled(true);

        // navController setup
        navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        NavigationUI.setupActionBarWithNavController(this, navController, drawerLayout);
        NavigationUI.setupWithNavController(navigationView, navController);
        navigationView.setNavigationItemSelectedListener(this);
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        item.setChecked(true);
        drawerLayout.closeDrawers();
        int id = item.getItemId();

//        switch (id) {
//            case R.id.navigation:
////                navController.navigate(R.id.navigationFragment);
//                break;
//            case R.id.view_model:
////                navController.navigate(R.id.viewModelFragment);
//                break;
//            case R.id.broadcast_receiver:
////                navController.navigate(R.id.broadcastReceiverFragment);
//                break;
//            case R.id.facebook_google:
////                navController.navigate(R.id.socialLoginFragment);
//                break;
//        }
        return true;
    }
}

File: app/src/main/java/com/fisherman/ordertogether/MainActivity.kt
package com.fisherman.ordertogether

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.ui.NavigationUI
import com.google.android.material.navigation.NavigationView
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.toolbar.*

class MainActivity : AppCompatActivity() {

    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupNavigation()
        setupListeners()
    }

    private fun setupNavigation() { // attaching view components to view variables
        // toolbar setup
        setSupportActionBar(toolbar)    //
        supportActionBar?.setDisplayShowHomeEnabled(true)   // show the home button
        supportActionBar?.setDisplayHomeAsUpEnabled(true)   // enable home button to become an up button

        // navController setup
        navController = Navigation.findNavController(this, R.id.nav_host_fragment)
        NavigationUI.setupActionBarWithNavController(this, navController, drawerLayout)
        NavigationUI.setupWithNavController(navigationView, navController)
    }

    private fun setupListeners() {
        navigationView.setNavigationItemSelectedListener { item ->
            when (item.itemId) {
                // TODO add more navigation destinations here
                else -> true
            }
        }
    }

    override fun onSupportNavigateUp() = NavigationUI.navigateUp(navController, drawerLayout)

    override fun onBackPressed() {
        when (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            true -> drawerLayout.closeDrawer(GravityCompat.START)
            false -> super.onBackPressed()
        }
    }
}